teemaderegister-be/src/controllers/csv.js

```diff
@@ -0,0 +1,78 @@
+const Topic = require('../models/topic')
+
+module.exports.findCsvData = async (req, res) => {
+  const { status, course } = req.query
+  switch (status) {
+    case 'registered':
+      await Topic.find({
+        'curriculums': course,
+        'defended': { $exists: false },
+        'registered': {$exists: true}},
+      (err, docs) => {
+        if (!err) {
+          return docs
+        } else {
+          throw err
+        }
+      })
+        .sort({registered: 'desc'})
+        .populate('supervisors.supervisor', '_id profile.firstName profile.lastName')
+        .populate('curriculums')
+        .select('title author registered curriculums')
+        .exec((err, data) => {
+          if (!err) {
+            res.send(JSON.stringify(data))
+          } else {
+            console.log(err)
+          }
+        })
+      break
+    case 'available':
+      await Topic.find({
+        'curriculums': course,
+        'defended': {$exists: false},
+        'registered': {$exists: false}},
+      (err, docs) => {
+        if (!err) {
+          return docs
+        } else {
+          throw err
+        }
+      })
+        .sort({accepted: 'desc'})
+        .populate('supervisors.supervisor', '_id profile.firstName profile.lastName')
+        .populate('curriculums')
+        .select('title createdAt curriculums')
+        .exec((err, data) => {
+          if (!err) {
+            res.send(JSON.stringify(data))
+          } else {
+            console.log(err)
+          }
+        })
+      break
+    case 'defended':
+      await Topic.find({
+        'curriculums': course,
+        'defended': {$exists: true}},
+      (err, docs) => {
+        if (!err) {
+          return docs
+        } else {
+          throw err
+        }
+      })
+        .sort({defended: 'desc'})
+        .populate('supervisors.supervisor', '_id profile.firstName profile.lastName')
+        .populate('curriculums')
+        .select('title author defended curriculums')
+        .exec((err, data) => {
```

Comment:
I would try to optimise this large switch

it is possible to construct the query dynamically, e.g
```
cons query = {}
let sort;
/// switch case.... query.defended =  {$exists: false} ... sort = {defended: 'desc'}
await Topic.find(query).sort(sort)
...

```